pandas/_libs/tslibs/conversion.pyx

```diff
@@ -928,11 +933,16 @@ def tz_localize_to_utc(ndarray[int64_t] vals, object tz, object ambiguous=None,
 
     if nonexistent == 'NaT':
         fill_nonexist = True
-    elif nonexistent == 'shift':
-        shift = True
-    else:
-        assert nonexistent in ('raise', None), ("nonexistent must be one of"
-                                                " {'NaT', 'raise', 'shift'}")
+    elif nonexistent == 'shift_forward':
+        shift_forward = True
+    elif nonexistent == 'shift_backward':
+        shift_backward = True
+    elif PyDelta_Check(nonexistent):
```

Comment:
is there a practical use of this?

Reply:
From the original enhancement request https://github.com/pandas-dev/pandas/issues/24466#issue-394621856, there was no way to shift backwards, moreover by a specified amount. I can see how it would be valuable to shift nonexistent times forward or backwards by a specified amount instead of the closest time. (e.g. I want times on the half hours, so be able to shift times to 1:30 or 3:30 and not just 1:59 or 3:00)

Reply:
is this duplicating shift_backwards though? this seems unecessarily complex to do anything more that doing a snap, eg. to a valid time (forward or backwards). shifting is an independent operation.

Reply:
IMO it didn't add _too_ much more complexity to the code, but here was @sdementen use case:

> I do some calculation on a local timestamp without tz, that I shift by 2 hours backward (to say "take it two hours before") and then I localize and get a NonExistentTimeError (e.g. Timestamp("2018-03-25T04:33:00") - DateOffset(hours=2)). I would like to get as a result of the tz_localize('CET'), the time "2018-03-25T01:33:00+0100" or "2018-03-25T03:33:00+0200" (and not "2018-03-25T01:59:59.99999+0100" or "2018-03-25T03:00:00+0200")

Reply:
I can see how in principle how it's very similar to `shift_forward`/ `shift_backwards` then adding an offset, but it can be easier determining how you want to shift from a nonexistent time compared to a nonexistent time that has been snapped to 1:59 or 3:00.

Reply:
@sdementen care to chime in on thoughts?